cache/in-memory/src/event/mod.rs

```diff
@@ -67,20 +67,14 @@ impl UpdateCache for Ready {
             cache.cache_current_user(self.user.clone());
         }
 
-        if cache.wants(ResourceType::GUILD) {
-            for guild in &self.guilds {
-                cache.unavailable_guild(guild.id);
-            }
+        for guild in &self.guilds {
+            cache.unavailable_guild(guild.id);
```

Comment:
I believe I'm getting tripped up on why we're always calling `unavailable_guild` here but are continuing to gate it in `cache_guild` and `delete_guild`. This reads to me like we're causing all guilds to be left in an unavailable state indefinitely?

Reply:
If you mean the `unavailable_guilds` set there, the `cache_guild` function removes the `guild_id` from there when it gets called.

Reply:
I think we can just leave the `if cache.wants(ResourceType::GUILD) {` check here? We can still cache presences and members separately and remove them on guild deletes without checking for the GUILD resource type being enabled, I think we just shouldn't cache (unavailable) guilds themselves if guilds aren't wanted

Reply:
I think this is fine + keeping it on `UnavailableGuild`

Reply:
I would do that tomorrow. But what's the final decision on this issue?